src/app/location-map/location-map.component.html

```diff
@@ -1,3 +1,3 @@
-<div class="map-container">
+<div class="map-container" [class.full-height]="fullHeight">
```

Comment:
Can't you just set the height to the inner map element to span the maximum height, so you can control the height from outside without this variable?

Reply:
Like i thought, issue is with Angular encapsulation. This is why we can't edit child component styles from parent component.
Go to move to fix this used to be to use ::ng-deep but this is deprecated.
Instead I found [this](https://stackoverflow.com/a/50159982). It removes encapsulation from parent component.
This way styles become global!! To constrain styles to component only, we then add container to the component with unique class and move all CSS in that class.
I tried this and it seems to be working fine. Didn't see any styles being overwritten as well but to make sure QA should quickly go through app and check if anything is broken.